File: ung_dung_thu_y/lib/dto/user/user_creation_request.dart
class UserCreationRequest {
  final String fullname;
  final String birthday;
  final String phoneNumber;
  final int gender;
  final String address;

  UserCreationRequest({
    required this.fullname,
    required this.birthday,
    required this.phoneNumber,
    required this.gender,
    required this.address,
  });

  Map<String, dynamic> toJson() {
    return {
      'fullname': fullname,
      'birthday': birthday,
      'phone_number': phoneNumber,
      'gender': gender,
      'address': address,
    };
  }

  factory UserCreationRequest.fromJson(Map<String, dynamic> json) {
    return UserCreationRequest(
      fullname: json['fullname'],
      birthday: json['birthday'],
      phoneNumber: json['phoneNumber'],
      gender: json['gender'],
      address: json['address'],
    );
  }
}

File: be/src/main/java/com/example/petclinic/dto/request/user/UserCreationRequest.java
package com.example.petclinic.dto.request.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
@AllArgsConstructor
public class UserCreationRequest {
    String fullname;
    String birthday;
    @JsonProperty("phone_number")
    String phoneNumber;
    String email;
    int gender;
    String address;
}
